aws_lambda_builders/workflows/nodejs_npm_esbuild/esbuild.py

```diff
@@ -102,3 +95,198 @@ def run(self, args, cwd=None):
+SUPPORTED_ESBUILD_APIS_SINGLE_VALUE = [
+    "target",
+]
+
+# Multi-value types (--external:axios --external:aws-sdk)
+SUPPORTED_ESBUILD_APIS_MULTI_VALUE = [
+    "external",
+    "loader",
+]
+
+
+class EsbuildCommandBuilder:
+    ENTRY_POINTS = "entry_points"
+
+    def __init__(
+        self, scratch_dir: str, artifacts_dir: str, bundler_config: Dict[Any, Any], osutils: OSUtils, manifest: str
+    ):
+        self._scratch_dir = scratch_dir
+        self._artifacts_dir = artifacts_dir
+        self._bundler_config = bundler_config
+        self._osutils = osutils
+        self._manifest = manifest
+        self._command = []
+
+    def get_command(self) -> List[str]:
+        """
+        Get all of the commands flags created by the command builder
+
+        :rtype: List[str]
+        :return: List of esbuild commands to be executed
+        """
+        return self._command
+
+    def build_esbuild_args_from_config(self) -> "EsbuildCommandBuilder":
+        """
+        Build arguments configured in the command config (e.g. template.yaml)
+
+        :rtype: EsbuildCommandBuilder
+        :return: An instance of the command builder
+        """
+        args = []
+
+        args.extend(self._get_boolean_args())
+        args.extend(self._get_single_value_args())
+        args.extend(self._get_multi_value_args())
+
+        LOG.debug("Found the following args in the config: %s", str(args))
+
+        self._command.extend(args)
+        return self
+
+    def build_entry_points(self) -> "EsbuildCommandBuilder":
+        """
+        Build the entry points to the command
+
+        :rtype: EsbuildCommandBuilder
+        :return: An instance of the command builder
+        """
+        if self.ENTRY_POINTS not in self._bundler_config:
+            raise EsbuildCommandError(f"{self.ENTRY_POINTS} not set ({self._bundler_config})")
+
+        entry_points = self._bundler_config[self.ENTRY_POINTS]
+
+        if not isinstance(entry_points, list):
+            raise EsbuildCommandError(f"{self.ENTRY_POINTS} must be a list ({self._bundler_config})")
+
+        if not entry_points:
+            raise EsbuildCommandError(f"{self.ENTRY_POINTS} must not be empty ({self._bundler_config})")
+
+        entry_paths = [self._osutils.joinpath(self._scratch_dir, entry_point) for entry_point in entry_points]
+
+        LOG.debug("NODEJS building %s using esbuild to %s", entry_paths, self._artifacts_dir)
+
+        for entry_path, entry_point in zip(entry_paths, entry_points):
+            self._command.append(self._get_explicit_file_type(entry_point, entry_path))
+
+        return self
+
+    def build_default_values(self) -> "EsbuildCommandBuilder":
+        """
+        Build the default values that each call to esbuild should contain
+
+        :rtype: EsbuildCommandBuilder
+        :return: An instance of the command builder
+        """
+        args = ["--bundle", "--platform=node", "--format=cjs", "--outdir={}".format(self._artifacts_dir)]
```

Comment:
Are we planning to parameterize the `platform` and `format` flags as well?

Reply:
I've been debating this myself. I'm not sure when/if platform would change, but since Lambda now allows for ES modules so format could be configurable.

Reply:
I parameterized `format` as well.